### PR TITLE
fix: Update the release process

### LINKED ISSUE #139: Warning: a recent release failed
The following release PRs may have failed:

* #124 - The release job is 'autorelease: tagged', but expected 'autorelease: published'.

### PR DESCRIPTION
**Pull request by averikitsch**
[go/java-cloud-library-playbook](https://goto.google.com/java-cloud-library-playbook) mentions Java needs:
* the release trigger app
* kokoro release configs. 

The label "autorelease: published" gets added by the release trigger app.

### CODE REVIEW

File: .kokoro/release/stage.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+# Start the releasetool reporter
+python3 -m pip install gcp-releasetool
+python3 -m releasetool publish-reporter-script > /tmp/publisher-script; source /tmp/publisher-script
```
Comment (chingor13) [inline .kokoro/release/stage.sh:20]: This file looks out of date. We require loading from a lockfile. example: https://github.com/googleapis/google-http-java-client/blob/0664e1744e0885a1cb8787481ccfbab0de845fe9/.kokoro/release/stage.sh#L19-L21

**Review by chingor13 — CHANGES_REQUESTED** [review]
Can you ensure that the .sh files are in executable mode as well?

File: .kokoro/linkage-monitor.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+retry_with_backoff 3 10 \
+  mvn install -B -V \
+    -DskipTests=true \
+    -Dclirr.skip=true \
+    -Denforcer.skip=true \
+    -Dmaven.javadoc.skip=true \
+    -Dgcloud.download.skip=true
+
+# Kokoro job cloud-opensource-java/ubuntu/linkage-monitor-gcs creates this JAR
+JAR=linkage-monitor-latest-all-deps.jar
```
Comment (meltsufin) [inline .kokoro/linkage-monitor.sh:42]: Do you really need the linkage-monitor?

Comment (suztomo) [reply .kokoro/linkage-monitor.sh:42]: No.

**Comment by averikitsch** [comment]
@chingor13 can you rereview? I've pulled a new copy of the Kokoro release files and have filed an issue to explore Owlbot in the future #142

**Review by chingor13 — APPROVED** [review]

**Review by muncus — APPROVED** [review]

**Pull request merged**